### ISSUE TITLE
Tests associated with filters

### ISSUE THREAD

**Issue by stevelinton** [gap-system/gap#532]
An interesting idea from SAGE -- we could have tests (simple ones I imagine) associated with filters. For instance IsAssociative might test associativity explicitly for some PseudoRandom elements.  Then given any object you could ask GAP to run the tests for all the filters it had. 

The most obvious application is when you are implementing a new kind of object and you want to gain reassurance that it hasn't acquired any filters that it doesn't deserve.

**Comment by fingolfin** [comment]
Yeah, I think that would be useful. We planned something like that for `SingularInterface` (but never really quite finished it); this is a special instance of `fuzzy` testing, where you create a bunch of random objects of a certain type, and feed them into operations. This can uncover a surprising amount of bugs. Of course it is no replacement for carefully handcrafted tests, but it supplements those quite well (and, in their absence, is much better than nothing)
